src/cocotb/share/include/gpi.h

```diff
@@ -153,132 +104,321 @@ typedef enum gpi_objtype_e {
+
+/** Stop the simulator. */
+GPI_EXPORT void gpi_sim_end(void);
 
-/**
- * @return `NULL` when there are no more objects.
+/** Get the simulation time as two uints. Unit is the default sim unit.
+ *
+ * @param high  Location to return high bits of current simulation time.
+ * @param low   Location to return low bits of current simulation time.
  */
-GPI_EXPORT gpi_sim_hdl gpi_next(gpi_iterator_hdl iterator);
+GPI_EXPORT void gpi_get_sim_time(uint32_t *high, uint32_t *low);
 
-/**
- * @return The number of objects in the collection of the handle.
+/** Get the simulation time precision.
+ *
+ * @param precision  Location to return time precision.
  */
-GPI_EXPORT int gpi_get_num_elems(gpi_sim_hdl gpi_sim_hdl);
+GPI_EXPORT void gpi_get_sim_precision(int32_t *precision);
 
-/**
- * @return The left side of the range constraint.
+/** Get the running simulator product information.
+ *
+ * @return The simulator product string.
+ */
+GPI_EXPORT const char *gpi_get_simulator_product(void);
+
+/** Get the running simulator version string.
+ *
+ * @return The simulator version string.
+ */
+GPI_EXPORT const char *gpi_get_simulator_version(void);
+
+/** @} */  // End of group SimIntf
+
+/** @defgroup ObjQuery Simulation Object Query
+ * These functions are for getting handles to simulation objects.
+ * @{
  */
-GPI_EXPORT int gpi_get_range_left(gpi_sim_hdl gpi_sim_hdl);
 
-/**
- * @return The right side of the range constraint.
+/** Get a handle to the root simulation object.
+ *
+ * @param name  Name of the root object.
+ * @return      Handle to simulation object or `NULL` if not found.
+ */
+GPI_EXPORT gpi_sim_hdl gpi_get_root_handle(const char *name);
+
+/** Get a handle to a child simulation object by its name.
+ *
+ * @param parent            Parent object handle.
+ * @param name              Name of the child object.
+ *                          This should not be a path,
+ *                          but only the name of a direct child object.
+ * @param discovery_method  Object discovery method.
+ * @return                  Handle to simulation object or `NULL` if not found.
  */
+GPI_EXPORT gpi_sim_hdl gpi_get_handle_by_name(gpi_sim_hdl parent,
+                                              const char *name,
+                                              gpi_discovery discovery_method);
+
+/** Get a handle to a child simulation object by its index.
+ *
+ * @param parent    Parent indexable object handle.
+ * @param index     0-based index of the child object.
+ * @return          Handle to simulation object or `NULL` if not found.
+ */
+GPI_EXPORT gpi_sim_hdl gpi_get_handle_by_index(gpi_sim_hdl parent,
+                                               int32_t index);
+
+/** @} */  // End of group ObjQuery
+
+/** @defgroup ObjProps General Object Properties
+ * These functions are for getting and setting properties of a simulation
+ * object.
+ * @{
+ */
+
+/** @return One of the GPI simulation object types defined above. */
```

Comment:
Maybe link the type?

Reply:
I tried a `@ref` with both `gpi_objtype_e` and `gpi_objtype` and both times the log said it couldn't find the reference. I'm not sure if it's related to the setting where the typedef hides the enum/struct names or not.